File: services/product/src/main/java/com/ecommerce/variant/VariantResponse.java
package com.ecommerce.variant;

import com.ecommerce.attributevalue.AttributeValueResponse;
import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

@Builder
public record VariantResponse (
        Long id,
        String sku,
        BigDecimal basePrice,
        BigDecimal discountedPrice,
        int quantity,
        boolean quantityUnlimited,
        Map<Long, AttributeValueResponse> attributeValues
) {}

File: store-ui/src/app/services/models/variant/variant-response.ts
import {AttributeValueResponse} from "../attribute/attribute-value-response";

export interface VariantResponse {
    id?: number;
    productId?: number;
    sku?: string;
    basePrice?: number;
    discountedPrice?: number;
    quantity?: number;
    quantityUnlimited?: boolean;
    attributeValues?: AttributeValueResponse[];
}
